cmd/protolock/main.go

```diff
@@ -98,16 +105,85 @@ func main() {
 
 	case "status":
 		report, err := protolock.Status(*ignore)
-		if err != nil {
-			handleReport(report, err)
+		if err != protolock.ErrWarningsFound && err != nil {
+			fmt.Println("[protolock]:", err)
+			os.Exit(1)
 		}
 
+		// if plugins are provided, attempt to execute each as a binary located
+		// in the user's OS executable filepath using exec.LookPath
+		if *plugins != "" {
+			report, err = runPlugins(*plugins, report)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+		}
+
+		handleReport(report, err)
+
 	default:
 		os.Exit(0)
 	}
 }
 
-func handleReport(report protolock.Report, err error) {
+func runPlugins(pluginList string, report *protolock.Report) (*protolock.Report, error) {
+	inputData := &bytes.Buffer{}
+
+	err := json.NewEncoder(inputData).Encode(&extend.Data{
+		Current:          report.Current,
+		Updated:          report.Updated,
+		ExistingWarnings: report.Warnings,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	plugins := strings.Split(pluginList, ",")
+	for i, name := range plugins {
```

Comment:
Is it possible to run plugins in parallel?

Reply:
I think it make sense in the case of `protoc` since one plugin may rely on the generated code of another plugin. However in the case of `protolock`, each check is independent of another. Thus I think we can try to parallelize it

Reply:
That's a good point. That gives user the ability to disable certain rules. It would be great if we can add an identifier field in `Warning` to indicate which rule they are violating so that users can easily filter out warnings based on the identifier